webapp/src/extensionsBrowser.tsx

```diff
@@ -0,0 +1,453 @@
+        workspace.getTextAsync(hd.id)
+            .then(files => {
+                let cfg = JSON.parse(files[pxt.CONFIG_NAME]) as pxt.PackageConfig
+                return addDepIfNoConflict(cfg, "workspace:" + hd.id)
+            })
+    }
+
+    function installExtension(scr: ExtensionMeta) {
+        switch (scr.type) {
+            case pxtc.service.ExtensionType.Bundled: {
+                pxt.tickEvent("packages.bundled", { name: scr.name });
+                props.hideExtensions();
+                addDepIfNoConflict(scr.pkgConfig, "*")
+            }
+            case pxtc.service.ExtensionType.Github: {
+                props.hideExtensions();
+                addGithubPackage(scr);
+            }
+                updateInstalledExts()
+        }
+    }
+
+    function parseGithubRepo(r: pxt.github.GitRepo): ExtensionMeta {
+        return {
+            name: r.name,
+            type: pxtc.service.ExtensionType.Github,
+            imageUrl: pxt.github.repoIconUrl(r),
+            repo: r,
+            description: r.description,
+            fullName: r.fullName
+        }
+    }
+
+
+    function getCategoryNames(): string[] {
+        return Array.from(extensionTags.keys())
+    }
+
+    function handleInstalledCardClick(src: ExtensionMeta) {
+        setDeletionCandidate(src)
+    }
+
+    async function handleCategoryClick(category: string) {
+        if (category == selectedTag) {
+            setSelectedTag("")
+            setExtensionsToShow([])
+            return;
+        }
+        setSelectedTag(category)
+        setSearchFor("")
+
+        const categoryExtensions = extensionTags.get(category)
+
+        const toBeFetched: string[] = []
+        const extensionsWeHave: ExtensionMeta[] = []
+
+        categoryExtensions.forEach(e => {
+            const fetched = getExtensionFromFetched(e.slug);
+            if (!fetched) {
+                toBeFetched.push(e.slug)
+            } else {
+                extensionsWeHave.push(fetched)
+            }
+        })
+
+        if (toBeFetched.length > 0) {
+            fetchGithubDataAndAddAsync(toBeFetched, (ext) => setExtensionsToShow([...extensionsWeHave, ...ext]))
+        }
+        const loadingCards = []
+        for (let i = 0; i < toBeFetched.length; i++) {
+            loadingCards.push(emptyCard)
+        }
+        setExtensionsToShow([...extensionsWeHave, ...loadingCards]);
+    }
+
+    function packageConfigToExtensionMeta(p: pxt.PackageConfig): ExtensionMeta {
+        return {
+            name: p.name,
+            imageUrl: p.icon,
+            type: pxtc.service.ExtensionType.Bundled,
+            pkgConfig: p,
+            description: p.description
+        }
+    }
+    function handleHomeButtonClick() {
+        setSelectedTag("")
+        setSearchFor("")
+    }
+
+    function fetchBundled(): Map<string, ExtensionMeta> {
+        const bundled = pxt.appTarget.bundledpkgs;
+        const extensionsMap = new Map<string, ExtensionMeta>();
```

Comment:
it's another real obj, but we don't really use it https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

maybe better to use `= {}: pxt.Map<string>`

Reply:
ohh is there....a reason we don't use it?

Reply:
no important ones, originally wasn't used due to inconsistent browser support / being a feature that was relatively new / unused but it's available everywhere now. just pretty very minor things

* since we use the Map type used all over the place it can be a little misleading at a glance -- doesn't really matter and easy to find when you're aware but if you e.g. pass it as a parameter or try to store it incorrectly as a pxt.Map the message is not ideal for debugging 
![image](https://user-images.githubusercontent.com/5615930/163658890-7c6cfe76-bdc1-462f-8aff-f0b290ab9ee8.png)
* whenever our existing functions use object literals they don't play well / have to convert to pass them in which is a bit magicky -- have to know to use `new Map<string, any>(Object.entries(obj))` for plain object -> map and `Object.fromEntries(myMap)` for map -> obj. 

which is not a big deal at all, just have to make sure everyone remembers that the map builtin type exists if we want to use it to avoid confusion there

Reply:
huh TIL